build-logic/convention/build.gradle.kts

```diff
@@ -37,7 +37,7 @@ kotlin {
 }
 
 dependencies {
-    compileOnly(libs.android.gradleApiPlugin)
+    compileOnly(libs.android.gradlePlugin)
```

Comment:
Can you share why we need to make this change?

Reply:
This is obviously not a required change, but I decided to remove (hide) the fact that is was an "api" version of the dependency since the "impl" is not supposed to be used in future releases. Keeping the distinction between the api and the impl visible in the dependency name is not that useful.